rails/app/controllers/api/v1/offerings_controller.rb

```diff
@@ -5,15 +5,26 @@
 class API::V1::OfferingsController < API::APIController
 
   def show
+    auth = auth_student_or_teacher_or_researcher(params)
+    return error(auth[:error]) if auth[:error]
+    user = auth[:user]
+
     offering = Portal::Offering
                    .where(id: params[:id])
                    .includes(API::V1::Offering::INCLUDES_DEF)
                    .first
     unless offering
       return error('offering not found', 404)
     end
-    authorize offering, :api_show?
```

Comment:
We probably want to eventually remove the `api_show?` pundit method in the offerings policy, unless we decide to move all the auth into pundit.

Reply:
I went the opposite way and moved everything to Pundit, as part of this PR: https://github.com/concord-consortium/rigse/pull/1257